lib/blockchain/migrations.js

```diff
@@ -0,0 +1,253 @@
+/*!
+ * blockchain/migrations.js - blockchain data migrations for hsd
+ * Copyright (c) 2021, Nodari Chkuaselidze (MIT License).
+ * https://github.com/handshake-org/hsd
+ */
+
+'use strict';
+
+const assert = require('bsert');
+const Logger = require('blgr');
+const Network = require('../protocol/network');
+const rules = require('../covenants/rules');
+const layout = require('./layout');
+const {
+  AbstractMigration,
+  Migrations,
+  types
+} = require('../migrations/migrations');
+
+/**
+ * Migrate chain state and correct total supply.
+ * Applies to ChainDB v1
+ */
+
+class MigrateChainState extends AbstractMigration {
```

Comment:
I think we talked about making this name more specific or linking to the github discussions somehow as metadata (this migration is old-migration 0 and 1 which are https://github.com/handshake-org/hsd/pull/396 and https://github.com/handshake-org/hsd/pull/585)

Reply:
Yep, that's one part of this migration that I have not touched so far. How to include information about the migration: whether it should be method so it can be called by API user or not. Have not decided yet.
Maybe properties ?
 ```
  this.issue = 396
  this.pr = 585
```
or have just a method that returns String description with links and small summary?

Reply:
Probably the `warning()` message is enough, like we did here: https://github.com/handshake-org/hsd/blob/master/lib/wallet/walletdb.js#L265-L268

I don't think `this.issue` or `this.pr` are great, but maybe an optional `this.description = <string>` which can mention issue or PR number.